lib/portlayer/exec/base.go

```diff
@@ -461,14 +454,77 @@ func (c *containerBase) waitForSession(ctx context.Context, id string) error {
 	return c.waitFor(ctx, key)
 }
 
-func (c *containerBase) waitForExec(op trace.Operation, id string) error {
+func (c *containerBase) WaitForExecStart(op trace.Operation, id string) error {
 	defer trace.End(trace.Begin(id, op))
 
 	// guestinfo key that we want to wait for
 	key := extraconfig.CalculateKeys(c.ExecConfig, fmt.Sprintf("Execs.%s.Started", id), "")[0]
 	return c.waitFor(op, key)
 }
 
+// Waits for the exec'd task to complete.
+//TODO: we really need structured Task style errors so that we can be more explicit in our handling upstream.
+func (c *containerBase) WaitForExec(op trace.Operation, id string) error {
+	op.Debugf("waiting for exec(%s) to complete", id)
+
+	// NOTE: Do we need to go to the property collector for all of these? The container base is from the exec cache... will it contain the new changes liked started?
+	startedKey := extraconfig.CalculateKeys(c.ExecConfig, fmt.Sprintf("Execs.%s.Started", id), "")[0]
```

Comment:
It is likely that we may not really need to go to extra config for `started` and `startTime` as we have the exec task info right there with the container base. Thoughts? 

Additionally, since the Wait func for exec config looks for changes in the exec config, it is possible that the exec finishes before we start waiting for changes. Thoughts/suggestions on this?